### PR TITLE
fix backup log always show

### PR DESCRIPTION
**Pull request by ldpliu**
https://issues.redhat.com/browse/ACM-9293
The kafka always sync the pvc label based on kafka cr pvc template, so we also need to update the template in kafka cr.

### CODE REVIEW

File: operator/pkg/controllers/backup/kafka.go [hunk 1]
```diff
@@ -47,7 +59,307 @@ func (r *kafkaBackup) AddLabelToOneObj(ctx context.Context,
...
+}
+
+func AddBackupLabelToKafkaTemplate(existingKafka *kafkav1beta2.Kafka) (*kafkav1beta2.Kafka, bool, error) {
+	var kafkaPVCLabels map[string]string
+	updatedKafka := existingKafka.DeepCopy()
+	if existingKafka == nil || existingKafka.Spec == nil {
+		return nil, false, fmt.Errorf("kafka spec should not be nil, name: %s}", existingKafka.Name)
+	}
+
+	if existingKafka.Spec.Kafka.Template == nil {
```
Comment (clyang82) [inline operator/pkg/controllers/backup/kafka.go:114]: You may define the desired template:
```
desiredTemplate := &kafkav1beta2.KafkaSpecKafkaTemplate{
			PersistentVolumeClaim: &kafkav1beta2.KafkaSpecKafkaTemplatePersistentVolumeClaim{
				Metadata: &kafkav1beta2.KafkaSpecKafkaTemplatePersistentVolumeClaimMetadata{
					Labels: &apiextensions.JSON{
						Raw: []byte(BackupVolumnLabelRaw),
					},
				},
			},
		}
```
Then
```
if existingKafka.Spec.Kafka.Template == nil {
    updatedKafka.Spec.Kafka.Template = desiredTemplate
}
if existingKafka.Spec.Kafka.Template.PersistentVolumeClaim == nil {
   updatedKafka.Spec.Kafka.Template.PersistentVolumeClaim = desiredTemplate.PersistentVolumeClaim
}
...
```

Comment (ldpliu) [reply operator/pkg/controllers/backup/kafka.go]: Good suggestion, fixed.

**Review by clyang82 — APPROVED** [review]
LGTM

**Pull request merged**